utils/src/test/scala/org/apache/hadoop/io/nativeio/NativeIO.java

```diff
@@ -0,0 +1,989 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.io.nativeio;
+
+// the file is copied from hadoop3.2.1, with just method org.apache.hadoop.io.nativeio.NativeIO.Windows.access changed
+// locally to make the tests run on Windows 10
+
```

Comment:
So is this file optional for other operating systems?

Reply:
Yes